src.ts/providers/provider-websocket.ts

```diff
@@ -38,66 +40,134 @@ export type WebSocketCreator = () => WebSocketLike;
  *  third-party services charge additional fees for WebSocket endpoints.
  */
 export class WebSocketProvider extends SocketProvider {
-    #connect: null | WebSocketCreator;
+    #websocket: WebSocketLike[];
+
+    readyMap: Map<string, boolean> = new Map();
 
-    #websocket: null | WebSocketLike;
-    get websocket(): WebSocketLike {
+    get websocket(): WebSocketLike[] {
```

Comment:
Since `websocket` will always refer to an array of websockets we should make it plural